libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -163,11 +160,7 @@ void DataFlowAnalyzer::handleAssignment(set<YulString> const& _variables, Expres
 
 	auto const& referencedVariables = movableChecker.referencedVariables();
 	for (auto const& name: _variables)
-	{
-		m_references[name] = referencedVariables;
```

Comment:
This is actually a change in semantics, but I believe it is correct.
The previous code just assigned `m_references[name] = referencedVariables` without previously clearing the old back-references. I still have to think of a test to check that, but it means that in
```
x := a
x := b
```
at the second assignment, `a` still thinks that it is referenced by `x`. This means if we assign to `a` again after that snippet, `x` should be cleared.

Reply:
A test could be:
```
x := a
x := 0
a := 9
mstore(x, 3)
```

Reply:
Ah no it's fine - the call to `clearValues` above in line 143/146 already does that.

Reply:
Still added the test.